lib/discourse_api/api/invite.rb

```diff
@@ -3,11 +3,30 @@ module DiscourseApi
   module API
     module Invite
       def invite_user(params = {})
-        post("/invites", params)
+        args = API.params(params)
+          .optional(
+            :email,
+            :skip_email,
+            :custom_message,
+            :max_redemptions_allowed,
+            :topic_id,
+            :group_ids,
+            :expires_at
+          ).to_h
+
+        post("/invites", args)
       end
 
-      def invite_user_to_topic(params = {})
-        post("/t/#{params[:topic_id]}/invite", params)
+      def invite_user_to_topic(topic_id, params = {})
```

Comment:
@oblakeerickson how's this looking now?

Reply:
Looking good! One thing I forgot to mention was that we want to avoid changing the method signatues for any methods that still call valid endpoints. So what we need to do actually is deprecate this method because we changed the signature and create a new method for these these changes. Check out these methods as an example of this:

https://github.com/discourse/discourse_api/blob/8bd6ef3e3c7bbf824bf37eb5672e0be92da70c23/lib/discourse_api/api/topics.rb#L50-L56

This way we don't introduce any breaking changes and we show a warning now that this method has been deprecated so that they have 1 year to update their code that is calling the deprecated method.